src/main/PresentationExchange.ts

```diff
@@ -138,47 +137,70 @@ export class PresentationExchange {
    * returns a PresentationDefinition if a valid instance found
    * @param obj: object that can have a presentation_definition inside
    */
-  public static findValidPresentationDefinitions(obj: JWTPayload): PresentationDefinitionWithLocation[] {
+  public static async findValidPresentationDefinitions(obj: JWTPayload): Promise<PresentationDefinitionWithLocation[]> {
     let allDefinitions: PresentationDefinitionWithLocation[];
-
-    function extractPDFromVPToken() {
+    async function extractPDFromVPToken() {
       const vpTokens = extractDataFromPath(obj, '$..vp_token.presentation_definition');
+      const vpTokenRef = extractDataFromPath(obj, '$..vp_token.presentation_definition_uri');
+      if (vpTokens && vpTokens.length && vpTokenRef && vpTokenRef.length) {
+        throw new Error(SIOPErrors.REQUEST_CLAIMS_CANT_SEND_PRESENTATION_DEFINITION_BY_RE_AND_VAL);
+      }
+      if (vpTokens && vpTokens.length == 1) {
```

Comment:
To make all of this a bit more readable suggest to change this to 

```
if (.....length ==1) {
...
} else if (....length > 1) {
....
} else if (ref && ref.length) {
...
}
```

Reply:
changed